From 3f3af7de3a918bc83e7523e50aa6b27711d5bd24 Mon Sep 17 00:00:00 2001
From: Albert Astals Cid <aacid@kde.org>
Date: Sat, 7 Jun 2025 12:37:22 +0200
Subject: [PATCH] Fix build with libgit2 1.9.1

The libgit2 headers are not correct and including blame.h for example
makes it failt to build because it doesn't know what
git_signature is because blame.h does not include types.h

So just include git2.h that includes everthing and be happy
---
 src/libkommit/blame.h                | 2 +-
 src/libkommit/blamehunk.h            | 2 +-
 src/libkommit/config.h               | 3 +--
 src/libkommit/options/blameoptions.h | 2 +-
 4 files changed, 4 insertions(+), 5 deletions(-)

diff --git a/src/libkommit/blame.h b/src/libkommit/blame.h
index 506a7623..bbef4c4f 100644
--- a/src/libkommit/blame.h
+++ b/src/libkommit/blame.h
@@ -6,7 +6,7 @@ SPDX-License-Identifier: GPL-3.0-or-later
 
 #pragma once
 
-#include <git2/blame.h>
+#include <git2.h>
 
 #include <QList>
 #include <QSharedPointer>
diff --git a/src/libkommit/blamehunk.h b/src/libkommit/blamehunk.h
index ada20f42..740da5e1 100644
--- a/src/libkommit/blamehunk.h
+++ b/src/libkommit/blamehunk.h
@@ -6,7 +6,7 @@ SPDX-License-Identifier: GPL-3.0-or-later
 
 #pragma once
 
-#include <git2/blame.h>
+#include <git2.h>
 
 #include <Kommit/Commit>
 #include <Kommit/Oid>
diff --git a/src/libkommit/config.h b/src/libkommit/config.h
index d09c6b8e..165394f5 100644
--- a/src/libkommit/config.h
+++ b/src/libkommit/config.h
@@ -9,8 +9,7 @@ SPDX-License-Identifier: GPL-3.0-or-later
 #include <QSharedPointer>
 #include <QString>
 
-#include <git2/config.h>
-#include <git2/version.h>
+#include <git2.h>
 
 namespace Git
 {
diff --git a/src/libkommit/options/blameoptions.h b/src/libkommit/options/blameoptions.h
index d4e65ab4..9e6103f7 100644
--- a/src/libkommit/options/blameoptions.h
+++ b/src/libkommit/options/blameoptions.h
@@ -10,7 +10,7 @@ SPDX-License-Identifier: GPL-3.0-or-later
 #include <QScopedPointer>
 #include <QSharedPointer>
 
-#include <git2/blame.h>
+#include <git2.h>
 
 namespace Git
 {
-- 
2.51.1

